sbtc-core/src/signer/config.rs

```diff
@@ -1,16 +1,245 @@
-use bitcoin::Address as BitcoinAddress;
-use secp256k1::PublicKey;
+use bitcoin::{Address as BitcoinAddress, Network};
+use serde::Deserialize;
 use stacks_core::address::StacksAddress;
+use std::str::FromStr;
+use toml;
+
+use crate::{SBTCError, SBTCResult};
 
 #[derive(Clone, Debug)]
 /// Configuration for the signer approval/denial
 pub struct Config {
-    /// The maximum dollar amount of a transaction that will be auto approved
-    pub auto_approve_max_amount: u64,
-    /// The public key of the signer being delegated to
-    pub delegate_public_key: PublicKey,
+    /// The maximum dollar amount of a transaction that will be auto approved.
+    pub auto_approve_max_amount: Option<u64>,
     /// The BTC addresses to be auto denied
     pub auto_deny_addresses_btc: Vec<BitcoinAddress>,
     /// The STX addresses to be auto denied
     pub auto_deny_addresses_stx: Vec<StacksAddress>,
+    /// The number of blocks before deadline at which point the transaction will be auto denied. Default is 10 blocks.
+    pub auto_deny_deadline_blocks: u32,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            auto_approve_max_amount: None,
+            auto_deny_addresses_btc: vec![],
+            auto_deny_addresses_stx: vec![],
+            auto_deny_deadline_blocks: 10,
+        }
+    }
+}
+
+impl Config {
+    /// Load the configuration from a toml file
+    pub fn from_path(path: impl AsRef<std::path::Path>) -> SBTCResult<Self> {
```

Comment:
This should probably have it's own result type. I don't think we should abuse the `SBTCResult` type and have it everywhere. This probably just is a IO result, or we can create a specialized custom result type. If you don't want to bother, I suggest defaulting to `anyhow` instead of `SBTCResult`.